perf: switch UUID default to gen_random_uuid

passcod opened the pull request:
### Changes

In #6326, it was found that a low-hanging fruit for sync performance was to change the serial generation from UUID to a bigserial.

Subsequently, [in the slack thread for the PR](https://beyondessential.slack.com/archives/C01RZFLB3DZ/p1724368895597569), we found that the `uuid_generate_v4()` function from the (built-in) extension `uuid-ossp` is **3.5 times slower** than `gen_random_uuid` from the (also built-in) extension `pgcrypto`, _specifically on Windows_. On Linux they're the same (and also performance for that one function is 30% greater on Linux given the same resources).

This PR migrates every single table and model to `gen_random_uuid`. It also adds default values for the `id` to a number of tables which had forgotten to do that (it doesn't touch tables with e.g. generated columns where the omission is intentional).

As we move to Linux this will become less important, but in the meantime this may provide some performance benefit to our Windows deployments.

passcod committed:
add pgcrypto extension

passcod committed:
pgcrypto: change models

passcod committed:
pgcrypto: migration

passcod committed:
pgcrypto: migration for varchar ids

passcod committed:
pgcrypto: migration for some one-offs

passcod committed:
pgcrypto: set the model default

passcod committed:
add lsf

passcod committed:
add many more no-defaults

passcod committed:
do migrations without changing types

passcod committed:
only add pgcrypto to <13

NavarroEmilioLuis reviewed (approved):
This LGTM but it leaves me with two questions:

- Is there any particular reason we don't iterate all tables in the same go, instead break them into categories? (string_ossp, uuid_ossp, string_no_default)
- Do we care/benefit changing types to one or the other?

passcod committed:
reduce duplication

passcod committed:
socket.io table is not present on facility

passcod committed:
fix migration checking for v12

passcod committed:
bump the last reversible migration

to something relatively significant ~2 years ago

we don't even support anything before 2.0, which was a year ago, there's zero need to go test way before then, but let's be conservative

passcod committed:
support migrating from scratch with the new code as well
